esp-hal/src/peripherals.rs

```diff
@@ -119,10 +119,29 @@ macro_rules! create_peripheral {
     };
 }
 
+macro_rules! io_type {
+    (Input, $peri:ident) => {
+        impl $crate::gpio::InputPin for $peri<'_> {
+            #[doc(hidden)]
+            #[inline(always)]
+            fn waker(&self) -> &'static $crate::asynch::AtomicWaker {
+                static WAKER: $crate::asynch::AtomicWaker = $crate::asynch::AtomicWaker::new();
+                &WAKER
+            }
+        }
+    };
+    (Output, $peri:ident) => {
+        impl $crate::gpio::OutputPin for $peri<'_> {}
+    };
+    ($other:ident, $peri:ident) => {
+        // TODO
```

Comment:
Nit: should we put a `todo!()` here instead?

Reply:
oh no this was supposed to go away long ago :D